src/main/app/controller/lists/LocalAuthoritiesListController.ts

```diff
@@ -0,0 +1,108 @@
+import {CSRF} from '../../../modules/csrf';
+import {AxiosError} from 'axios';
+
+
+@autobind
+export class LocalAuthoritiesListController {
+
+  getLocalAuthoritiesErrorMsg = 'A problem occurred when retrieving the list of local authorities. ';
+  updateErrorMsg = 'A problem occurred when saving the local authority. ';
+  emptyLocalAuthorityErrorMsg = 'The local authority name is required.';
+  duplicatedErrorMsg = 'Local Authority already exists. ';
+  invalidErrorMsg = 'Invalid Local Authority entered.';
+  notFoundErrorMsg = 'Local Authority not found.';
+
+
+  public async get(
+    req: AuthedRequest,
+    res: Response,
+    updated = false,
+    error = '',
+    updatedName = '',
+    selectedLocalAuthority: LocalAuthority = null): Promise<void> {
+
+    let allLocalAuthorities: LocalAuthority[] ;
+    await req.scope.cradle.api.getAllLocalAuthorities()
+      .then((value: LocalAuthority[]) => allLocalAuthorities = value)
+      .catch(() => error += this.getLocalAuthoritiesErrorMsg);
+
+    const pageData: LocalAuthoritiesListPageData = {
+      errorMsg: error,
+      updated: updated,
+      selected : selectedLocalAuthority,
+      updatedName: updatedName,
+      localAuthorities: this.mapLocalAuthorityToLocalAuthorityItem(allLocalAuthorities, selectedLocalAuthority)
+    };
+
+    res.render('lists/tabs/localAuthoritiesListContent', pageData);
+
+  }
+
+  public async put(req: AuthedRequest, res: Response): Promise<void> {
+
+    let selectedLocalAuthority: LocalAuthority;
+    const updatedName = req.body.localAuthorityName;
+
+    if (req.body.localAuthorities){
+      selectedLocalAuthority = JSON.parse(req.body.localAuthorities);
+    }
+
+    if(updatedName === '')
+    {
+      return this.get(req, res, false, this.emptyLocalAuthorityErrorMsg, updatedName, selectedLocalAuthority);
+    }
+
+    if(!CSRF.verify(req.body._csrf)) {
+      return this.get(req, res, false, this.updateErrorMsg,updatedName, selectedLocalAuthority);
+    }
+    else
+    {
+      await req.scope.cradle.api.updateLocalAuthority(selectedLocalAuthority.id , updatedName)
+        .then((value: LocalAuthority) => this.get(req, res, true, '',updatedName, value))
+        .catch(async (reason: AxiosError) => {
+          await this.get(req, res,false, this.returnResponseMessage(reason.response?.status), updatedName, selectedLocalAuthority);
+        });
+    }
+  }
+
+  private mapLocalAuthorityToLocalAuthorityItem(localAuthorities: LocalAuthority[], selectedLocalAuthority: LocalAuthority): LocalAuthorityItem[] {
+
+    if (localAuthorities) {
+
+      const localAuthorityItems = localAuthorities.map((la: LocalAuthority) => (
+        {
+          id: la.name.replace(/ /g,'_'),
+          value: JSON.stringify(la),
+          text: la.name,
+          checked : selectedLocalAuthority ? (la.id === selectedLocalAuthority.id) : false
+        }));
+
+      return localAuthorityItems.sort((a, b) => (a.text < b.text ? -1 : 1));
```

Comment:
Can we not use sort((a,b) => a.text - b.text)? This returns negative if a smaller than b and will return positive if a bigger than b. Does it not return the same result?

Reply:
tried using your suggestion return syntax "The right-hand side of an arithmetic operation must be of type 'any', 'number', 'bigint' or an enum type"